js/script.js

```diff
@@ -0,0 +1,95 @@
+const makersNewsSummaryApi = "http://news-summary-api.herokuapp.com/";
+const guardianApiRequestUrl = "guardian?apiRequestUrl=";
+const guardianPolitics = "http://content.guardianapis.com/politics";
+const guardian = "http://content.guardianapis.com/";
+const aylienApiRequestUrl =
+  "aylien?apiRequestUrl=https://api.aylien.com/api/v1/summarize?url=";
+const queryImageThumbnail = "?show-fields=thumbnail";
+
+for (let articleId = 0; articleId < 10; articleId++) {
+  fetchNewsArticle(articleId).then(
+    (data) =>
+      (document.getElementsByClassName("headline-title")[articleId].innerText =
+        data.webTitle)
+  );
+  fetchArticleImage(articleId);
```

Comment:
Here you are fetching 10 times for 10 articles. Is there a way to batch this request?

Reply:
I'm not sure I understand what you mean here. Do I not need to fetch 10 article headlines and 10 thumbnails for the page?

Reply:
You do, but you can get the 10 headlines to come back with the url to the thumbnail attached (see thumbnail field on this page: https://open-platform.theguardian.com/documentation/search). It still means you'll have to have more queries for the summary, but that can be delayed to only happen when the user clicks.